src/Tizen.Multimedia.Vision/MediaVision/InferenceImageClassifier.cs

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright (c) 2024 Samsung Electronics Co., Ltd All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the License);
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using InteropIC = Interop.MediaVision.InferenceImageClassification;
+
+namespace Tizen.Multimedia.Vision
+{
+    /// <summary>
+    /// Provides the ability to classify image.
+    /// </summary>
+    /// <feature>http://tizen.org/feature/vision.inference</feature>
+    /// <feature>http://tizen.org/feature/vision.inference.image</feature>
+    /// <since_tizen> 12 </since_tizen>
+    public class InferenceImageClassifier : IDisposable
+    {
+        private IntPtr _handle;
+        private bool _disposed;
+
+        /// <summary>Initializes a new instance of the <see cref="InferenceImageClassifier"/> class.</summary>
+        /// <exception cref="NotSupportedException">The required features are not supported.</exception>
+        /// <since_tizen> 12 </since_tizen>
+        public InferenceImageClassifier()
+        {
+            ValidationUtil.ValidateFeatureSupported(VisionFeatures.Inference);
+            ValidationUtil.ValidateFeatureSupported(VisionFeatures.InferenceImage);
+
+            InteropIC.Create(out _handle).Validate("Failed to create inference image classifier.");
+
+            try
+            {
+                InteropIC.Configure(_handle).Validate("Failed to configure inference image classifier.");
+                InteropIC.Prepare(_handle).Validate("Failed to prepare inference image classifier.");
+            }
+            catch (Exception e)
+            {
+                Log.Error(MediaVisionLog.Tag, e.ToString());
+                InteropIC.Destroy(_handle);
+                throw;
+            }
+        }
+
+        /// <summary>
+        /// Finalizes an instance of the InferenceImageClassifier class.
+        /// </summary>
+        ~InferenceImageClassifier()
+        {
+            Dispose(false);
+        }
+
+        /// <summary>
+        /// Classifies image objects on the source image synchronously.
+        /// </summary>
+        /// <remarks>
+        /// If there's no classified image, <see cref="InferenceImageClassifierResult.Labels"/> will be empty.
```

Comment:
Actually, it's not a classification fail.
it could be empty when there's no classifiable object in input source.  ex) all white, black source with some object
Or, it could be empty even if we cannot get label for input image that existing in internal label.